app/services/activities_service.rb

```diff
@@ -5,19 +5,40 @@ def self.load_activities(user, teams, filters = {})
     # Create condition for view permissions checking first
     visible_teams = user.teams.where(id: teams)
     visible_projects = Project.viewable_by_user(user, visible_teams)
+    visible_my_modules = MyModule.joins(:experiment)
```

Comment:
I think there is no need to join projects and experiments, MyModule.viewable_by_user(user, teams) should be enough